rmldatatfbbox/rmldatatfbbox/core.py

```diff
@@ -5,77 +5,158 @@
 Core command group and commands for TF Bounding Box dataset plugin.
 """
 import click 
-
-### OPTIONS ###
-
+import sys
+import io
+import importlib
+import shutil
+from pathlib import Path
+import tensorflow as tf
+from rmldatatfbbox.utils.helpers import (construct_all, write_label_map, export_as_TFExample)
+from ravenml.utils.question import user_input, cli_spinner
+from ravenml.data.helpers import default_filter_and_load
+from ravenml.data.write_dataset import write_dataset, write_metadata
+from ravenml.data.options import pass_create
+from ravenml.data.interfaces import CreateInput, CreateOutput
 
 ### COMMANDS ###
-@click.command(help='TensorFlow Object Detection with bounding boxes.')
+@click.group(help='Dataset creation with bounding box info')
 @click.pass_context
 def tf_bbox(ctx):
     pass
+
+@tf_bbox.command(help='Create a dataset.')
+@pass_create
+@click.pass_context
+def create(ctx: click.Context, create: CreateInput):
+
+    cli_spinner("Importing TensorFlow...", _import_od)
+
+    config = create.config.get("plugin")
+
+    # Optional Information from interface (just contains plugin)
+    # metadata = create.plugin_metadata
+
+    # set up TF verbosity
+    if config.get('verbose'):
+        tf.compat.v1.logging.set_verbosity(tf.compat.v1.logging.INFO)
+    else:
+        tf.compat.v1.logging.set_verbosity(tf.compat.v1.logging.FATAL)
     
-# @tf_bbox.command(help='Create a dataset.')
-# def create(ctx):
-#     click.echo("Dataset gets created here.")
-
-# # stdout redirection found at https://codingdose.info/2018/03/22/supress-print-output-in-python/
-# def _import_od():
-#     """ Imports the necessary libraries for object detection training.
-#     Used to avoid importing them at the top of the file where they get imported
-#     on every ravenML command call, even those not to this plugin.
+    # set base directory for dataset 
+    base_dir = create.dataset_path
     
-#     Also suppresses warning outputs from the TF OD API.
-#     """
-#     # create a text trap and redirect stdout
-#     # to suppress printed warnings from object detection and tf
-#     text_trap = io.StringIO()
-#     sys.stdout = text_trap
-#     sys.stderr = text_trap
+    # Transformation functionality is not currently supported, thus the empty array
+    transform_metadata = []
+
+    # Variable type validation is currently not supported
+    dataset_name = config.get("dataset_name") if config.get("dataset_name") else user_input(message="What would you like to name this dataset?")
```

Comment:
This is functionality that should be placed into the `__init__` for the `CreateInput`, see same lines as I referenced above (https://github.com/autognc/ravenML/blob/master/ravenml/train/interfaces.py#L44)